File: 724-find-pivot-index/find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        left_sum = 0
        i = 0
        sum = 0
        left =0 
        while i < len(nums):
            sum += nums[i]
            i +=1
        for k in range(len(nums)):
            while left < k:
                left_sum += nums[left]
                left +=1
            if sum - left_sum - nums[k] == left_sum:
                return k
            left_sum =0
            left = 0
        return -1

File: 724-find-pivot-index/find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int sum = 0;
        for(int i : nums){
            sum+= i;
        }
        int leftSum = 0;
        for(int i = 0;i< nums.length;i++){
            if(i>0){
                leftSum += nums[i-1];
                sum = sum - nums[i-1]; 
            }
            if(leftSum == (sum - nums[i]) ){
                return i;
            }
        }
        return -1;
    }
}